test/sqllogictest/cluster.slt

```diff
@@ -188,3 +188,13 @@ DROP CLUSTER baz CASCADE
 query T
 SELECT name FROM mz_indexes;
 ----
+
+# Test that dropping a cluster and re-creating it with the same name is valid
```

Comment:
@umanwizard This actually wasn't failing on main, is that because introspection sources aren't enabled for SLT tests?

Reply:
yep, check #11791 . We are actually not enabling introspection sources for clusters at all right now by default, and 11791 does so. That's how I discovered this issue in the first place.